src/tss2-fapi/ifapi_eventlog_system.c

```diff
@@ -0,0 +1,766 @@
+        return false;
+    }
+
+    TCG_VENDOR_INFO *vendor = (TCG_VENDOR_INFO*)((uintptr_t)event_specid->digestSizes +
+                                                 sizeof(*event_specid->digestSizes) *
+                                                 event_specid->numberOfAlgorithms);
+    /* size must be sufficient for vendorInfo */
+    if (size < sizeof(*event) + sizeof(*event_specid) +
+               sizeof(event_specid->digestSizes[0]) *
+               event_specid->numberOfAlgorithms + sizeof(*vendor) +
+               vendor->vendorInfoSize) {
+        LOG_ERROR("insufficient size for VendorStuff data");
+        return false;
+    }
+    *next = (TCG_EVENT_HEADER2*)((uintptr_t)vendor->vendorInfo + vendor->vendorInfoSize);
+
+    return true;
+}
+
+static TCG_EVENT_TYPE_ASSIGN deserialize_TCG_EVENT_TYPE_tab[] = {
+    { EV_PREBOOT_CERT, "EV_PREBOOT_CERT" },
+    { EV_POST_CODE, "EV_POST_CODE" },
+    { EV_UNUSED, "EV_UNUSED" },
+    { EV_NO_ACTION, "EV_NO_ACTION" },
+    { EV_SEPARATOR, "EV_SEPARATOR" },
+    { EV_ACTION, "EV_ACTION" },
+    { EV_EVENT_TAG, "EV_EVENT_TAG" },
+    { EV_S_CRTM_CONTENTS, "EV_S_CRTM_CONTENTS" },
+    { EV_S_CRTM_VERSION, "EV_S_CRTM_VERSION" },
+    { EV_CPU_MICROCODE, "EV_CPU_MICROCODE" },
+    { EV_PLATFORM_CONFIG_FLAGS, "EV_PLATFORM_CONFIG_FLAGS" },
+    { EV_TABLE_OF_DEVICES, "EV_TABLE_OF_DEVICES" },
+    { EV_COMPACT_HASH, "EV_COMPACT_HASH" },
+    { EV_IPL, "EV_IPL" },
+    { EV_IPL_PARTITION_DATA, "EV_IPL_PARTITION_DATA" },
+    { EV_NONHOST_CODE, "EV_NONHOST_CODE" },
+    { EV_NONHOST_CONFIG, "EV_NONHOST_CONFIG" },
+    { EV_NONHOST_INFO, "EV_NONHOST_INFO" },
+    { EV_OMIT_BOOT_DEVICE_EVENTS, "EV_OMIT_BOOT_DEVICE_EVENTS" },
+    { EV_EFI_VARIABLE_DRIVER_CONFIG, "EV_EFI_VARIABLE_DRIVER_CONFIG" },
+    { EV_EFI_VARIABLE_BOOT, "EV_EFI_VARIABLE_BOOT" },
+    { EV_EFI_BOOT_SERVICES_APPLICATION, "EV_EFI_BOOT_SERVICES_APPLICATION" },
+    { EV_EFI_BOOT_SERVICES_DRIVER, "EV_EFI_BOOT_SERVICES_DRIVER" },
+    { EV_EFI_RUNTIME_SERVICES_DRIVER, "EV_EFI_RUNTIME_SERVICES_DRIVER" },
+    { EV_EFI_GPT_EVENT, "EV_EFI_GPT_EVENT" },
+    { EV_EFI_ACTION, "EV_EFI_ACTION" },
+    { EV_EFI_PLATFORM_FIRMWARE_BLOB, "EV_EFI_PLATFORM_FIRMWARE_BLOB" },
+    { EV_EFI_HANDOFF_TABLES, "EV_EFI_HANDOFF_TABLES" },
+    { EV_EFI_HCRTM_EVENT, "EV_EFI_HCRTM_EVENT" },
+    { EV_EFI_VARIABLE_AUTHORITY, "EV_EFI_VARIABLE_AUTHORITY" },
+};
+
+/**  Deserialize a json object of type TCG_EVENT_TYPE.
+ *
+ * @param[in]  jso the json object to be deserialized.
+ * @param[out] out the deserialzed binary object.
+ * @retval TSS2_RC_SUCCESS if the function call was a success.
+ * @retval TSS2_FAPI_RC_BAD_VALUE if the json object can't be deserialized.
+ */
+TSS2_RC
+ifapi_json_TCG_EVENT_TYPE_deserialize_txt(json_object *jso,
+        IFAPI_EVENT_TYPE *out)
+{
+    LOG_TRACE("call");
+    const char *token = json_object_get_string(jso);
+    int64_t i64;
+    if (get_number(token, &i64)) {
+        *out = (IFAPI_EVENT_TYPE) i64;
```

Comment:
Why have get_number return an i64 when out is bounded on UINT32? Why not just use strtol and get a uint32_t?

Reply:
I did always use int64_t because we did serialize unit32 with  json_object_new_int64.